0000-interledger-quoting-protocol-v2.md

```diff
@@ -0,0 +1,87 @@
+4. When the receiver is notified of the incoming transfer, they reject the transfer with an error message including the amount that arrived in the transfer and the amount of time they have before the transfer expires.
+5. Connectors relay the error message back to the sender.
+6. When the sender is notified that their test transfer was rejected, they inspect the error message to determine how much arrived at the receiver's ledger and how much time the receiver had to respond before the transfer would time out. The sender can then adjust the values before sending the actual payment.
+
+### Fixed Destination Amount
+
+The steps are the same as for Fixed Source Amount quotes, except that the transfer amount is set to an arbitrary probe value. When the sender gets the error response from the receiver, they compute the exchange rate based on the amount they sent and the amount that arrived. The sender then divides the fixed destination amount from the quote request by that exchange rate.
+
+## Specification
+
+### Condition
+
+ILQPv2 test payments use the condition:
+
+`fffffffffffffff8a5aa9fefdbed3fffffffffffffffffffffffffffffffffff`, encoded as hexadecimal.
+
+Or `__________ilqp_v2-0________________________`, encoded in Base64-URL.
+
+### Error Message Format
+
+**TODO: Define a binary format**
+
+Fields include:
+- Error code: either `F08` (which is currently unused) or `F99: Application Error`
+- Error name: `ILQPv2`
+- Amount that arrived in the final transfer to the receiver
+- Milliseconds between the final transfer expiry and when the receiver processed the notification
+
+## Open Questions
+
+1. Should this be a standalone protocol or should it be built into Transport Layer protocols like PSK? One advantage of building it into something like PSK is that in that case you could use the shared secret to authenticate and encrypt the quote message.
+2. Will senders want to hide the fact that they are requesting a quote from connectors (for example by using data in the ILP Packet as opposed to a well-known condition to indicate that the payment meant as a quote request)? Connectors may have good reasons (such as DoS prevention) for wanting to avoid having lots of failed payments going through them. On the other hand, using a known condition could allow connectors that recognize this protocol to avoid placing funds on hold.
+3. Should we try to figure out quoting for streaming payments now or is this fine for the moment?
+4. How should the sender determine the expiry duration for the transfer they send? How likely are we to run into very slow ledgers? Should they have logic for increasing the timeout if necessary? They might not want to put the transfer on hold for too long if their ledger actually requires them to put their funds on hold (as opposed to recognizing that this is meant just as a quote request and won't be fulfilled).
```

Comment:
The sender can observe how long the quote request took. In the case of Ethereum and XRP Escrow the sender would observe that the quote request took 15 seconds or 3.7 seconds respectively. It could assume that the fulfillment path takes about as long as the preparation path and submit the payment with a timeout of 2.5x the time that the quote took.

I'm not too concerned about supporting escrow plugins because I think we will quickly evolve towards fast ledgers / payment channels. At that point we may move to having a constant timeout (e.g. 60 seconds) most of the time. (Interplanetary ILP will be a fun project, but perhaps a bit premature to design around that use case right now.)

Reply:
I meant how long they should set the timeout for the quote request. If you set the expiry too soon the payment will timeout before it gets to the receiver. Do you send another one automatically with a higher expiry? What's the max you should set (/ what's the default we should set in the `ilp` module)?